platform/engineering/request-for-comment/2022-02-23-remove-gender-fax-from-mobile-api

```diff
@@ -0,0 +1,23 @@
+RFC: Removing gender and fax number from mobile api
+Date: 2022-02-23
+Related Issue: department-of-veterans-affairs/va-mobile-app#2565
+Background
+Currently gender and fax are both returned from the mobile API and displayed in the app. 
+
+Motivation
+Profile is removing gender and fax number. More details: https://dsva.slack.com/archives/C018V2JCWRJ/p1643395405843779
+
+Design
+The proposed change for the API would be to add an accessor to the iam_user model to pull gender from the IAM identity or return null if it doesn't exist. This would remove the risk of the profile changes breaking the mobile API by making gender no longer accessible through the user model. Otherwise we propose doing nothing in the API until we are certain these fields have been completely removed from the upstream services and confident most users have updated the app beyond the version where the app no longer shows the fields.
```

Comment:
We can tie this back to the motivation. Our proposal is to version our api, removing gender and fax from the new version while, at the same time, supporting older versions of the app by getting that data (or at least gender) from somewhere else. That'll let Profile remove them.

Reply:
I'd like to at least provide some details here of when we'll be able to remove the fields entirely. I don't have a set # in mind, but once older versions of the app reach some small percentage, we'll be able to fully remove all references to those fields in vets-api. My guess is that small percentage will be reached in 6 to 8-ish months.

Reply:
The proposal in this doc doesn't actually require versioning the API. Yes users will see the empty state conditions, detailed [https://app.zenhub.com/workspaces/va-mobile-frontendqa-60f1a34998bc75000f2a489f/issues/department-of-veterans-affairs/va-mobile-app/2445](here), but my perspective is that's fine. The reason we want to add an accessor in iam_user is to insulate ourselves from how deeply gender would be removed. For example, if profile just removes it from the database then we don't actually need our own accessor, but if they delete the existing accessor in the base user in vets-api then that would break our existing code.